sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -233,6 +236,25 @@ class UDFSuite extends QueryTest with SharedSQLContext {
           """.stripMargin).toDF(), complexData.select("m", "a", "b"))
   }
 
+  test("don't collapse UDF in order to avoid recomputing it") {
```

Comment:
I have tried to do that. But as we need to use UDF, we can't do related import in the unit test such as `ProjectCollapsingSuite`. Any suggestion?